lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -1418,8 +1418,32 @@ SmallVector<Direction> PrimitiveOp::portDirections() {
 
 SmallVector<DictionaryAttr> PrimitiveOp::portAttributes() {
   SmallVector<DictionaryAttr> portAttributes;
-  for (size_t i = 0; i < getReferencedPrimitive().getAllPorts().size(); ++i)
-    portAttributes.push_back(DictionaryAttr());
+  OpBuilder builder(getContext());
+  auto cleanAttrs = [&](DictionaryAttr dict){
```

Comment:
I'd personally prefer this to be in its own static function with a comment explaining its behavior to avoid cluttering the portAttributes function.